File: src/RealWorld.Application/DTOs/CommentData.cs
using System.Text.Json.Serialization;

namespace RealWorld.Application.DTOs;

public class CommentData
{
    public string Id { get; set; } = string.Empty;
    public string Body { get; set; } = string.Empty;

    [JsonIgnore]
    public string ArticleId { get; set; } = string.Empty;

    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset UpdatedAt { get; set; }

    [JsonPropertyName("author")]
    public ProfileData ProfileData { get; set; } = new();
}

File: article-service/src/main/java/io/spring/articleservice/application/data/CommentData.java
package io.spring.articleservice.application.data;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import io.spring.articleservice.application.DateTimeCursor;
import io.spring.articleservice.application.Node;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.joda.time.DateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CommentData implements Node {
  private String id;
  private String body;
  @JsonIgnore private String articleId;
  private DateTime createdAt;
  private DateTime updatedAt;

  @JsonProperty("author")
  private ProfileData profileData;

  @Override
  public DateTimeCursor getCursor() {
    return new DateTimeCursor(createdAt);
  }
}
